finality-aleph/src/party/mod.rs

```diff
@@ -470,7 +472,54 @@ where
         };
         loop {
             let last_finalized_number = self.client.info().finalized_number;
-            debug!(target: "afa", "Highest finalized: {:?} session {:?}", last_finalized_number, session_id);
+            if last_finalized_number > first_block {
+                let next_session_id = SessionId(session_id.0 + 1);
+                let next_session_authorities = match self
+                    .client
+                    .runtime_api()
+                    .next_session_authorities(&BlockId::Number(first_block))
+                {
+                    Ok(authorities) => authorities
+                        .expect("authorities for next session must be available at first block of current session"),
+                    Err(e) => {
+                        error!(target: "afa", "Error when getting authorities for session {:?} {:?}", next_session_id, e);
+                        panic!("We didn't get the authorities for the session {:?}", next_session_id);
+                    }
+                };
+
+                let authority_verifier = AuthorityVerifier::new(next_session_authorities.clone());
+
+                match get_node_index(&authorities, self.keystore.clone()).await {
+                    Some(node_id) => {
+                        let authority_pen = AuthorityPen::new(
+                            next_session_authorities[node_id.0].clone(),
+                            self.keystore.clone(),
+                        )
+                        .await
+                        .expect("The keys should sign successfully");
+
+                        if let Err(e) = self.session_manager.start_validator_session(
```

Comment:
I would expect it to be necessary to call `start_validator_session` for session `K` in order to successfully finalize session `K`, but it seems to me that it is never called for the `starting_session` (the first session this Consensus Party handles). Is it correct?

Reply:
It is called once more, inside `spawn_authority_task`, when a given session is starting. So it is called